cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManagerTest.java

```diff
@@ -0,0 +1,337 @@
+import org.apache.kafka.clients.admin.DescribeClientQuotasResult;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.quota.ClientQuotaAlteration;
+import org.apache.kafka.common.quota.ClientQuotaEntity;
+import org.apache.kafka.common.quota.ClientQuotaFilter;
+import org.apache.kafka.common.quota.ClientQuotaFilterComponent;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static io.strimzi.operator.common.auth.TlsPemIdentity.DUMMY_IDENTITY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(VertxExtension.class)
+public class DefaultKafkaQuotasManagerTest {
+    private final static Long DEFAULT_PRODUCER_BYTE_RATE = 2000L;
+    private final static Long DEFAULT_CONSUMER_BYTE_RATE = 2000L;
+    private final static Double DEFAULT_MUTATION_RATE = 0.5;
+    private final static Integer DEFAULT_REQUEST_PERCENTAGE = 25;
+
+    private static Vertx vertx;
+    @SuppressWarnings(value = "unchecked")
+    private final ArgumentCaptor<List<ClientQuotaAlteration>> quotaAlternationToKafka = ArgumentCaptor.forClass(List.class);
+
+    @BeforeAll
+    public static void before() {
+        vertx = Vertx.vertx();
+    }
+
+    @AfterAll
+    public static void after() {
+        vertx.close();
+    }
+
+    /**
+     * Checks if the result of {@link DefaultKafkaQuotasManager#shouldAlterDefaultQuotasConfig(Reconciliation, Vertx, Admin, List, boolean)}
+     * when there are no default Kafka quotas set and none (or Strimzi Quotas plugin) are desired.
+     * The result of this check should be false.
+     *
+     * @param context   Vertx test context
+     */
+    @Test
+    void testShouldAlterQuotasWithNoneCurrentAndNotAKafkaQuotasPlugin(VertxTestContext context) {
+        // Mock the Admin client
+        Admin mockAdminClient = mock(Admin.class);
+
+        // Mock describing the current metadata version
+        mockDescribeQuotasResultEmpty(mockAdminClient);
+
+        List<ClientQuotaAlteration.Op> noneQuotas = DefaultKafkaQuotasManager.prepareQuotaConfigurationRequest(DefaultKafkaQuotasManager.emptyQuotasPluginKafka());
+
+        DefaultKafkaQuotasManager.shouldAlterDefaultQuotasConfig(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClient, noneQuotas, true)
+            .onComplete(context.succeeding(result -> {
+                context.verify(() -> assertFalse(result));
+                context.completeNow();
+            }));
```

Comment:
Would it make sense to write these test as we write most other tests? E.g.

```suggestion
        
        Checkpoint checkpoint = context.checkpoint();
        DefaultKafkaQuotasManager.shouldAlterDefaultQuotasConfig(Reconciliation.DUMMY_RECONCILIATION, vertx, mockAdminClient, noneQuotas, true)
            .onComplete(context.succeeding(result -> {
                assertThat(result, is(false));

                checkpoint.flag();
            }));
```

Reply:
okay, I saw that `assertFalse`, `assertTrue` etc. is used also in other tests. But I will change it to `assertThat`, thanks